sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -985,6 +985,23 @@ public void deleteDataset(String projectId, String datasetId)
           // impossible to insert into BigQuery, and so we send it out through the dead-letter
           // queue.
           if (nextRowSize >= maxRowBatchSize) {
+            Boolean isRetryAlways = false;
+            try {
+              // We verify whether the retryPolicy parameter is "retryAlways". If it is, then
+              // it will return true. Otherwise it will return false, or it may throw an NPE,
+              // which we need to catch and ignore.
+              isRetryAlways = retryPolicy.shouldRetry(null);
+            } catch (NullPointerException e) {
+              // We do not need to do anything about this exception, as it was expected;
+            }
+            if (isRetryAlways) {
+              throw new RuntimeException(
+                  String.format(
+                      "We have observed a row that is %s bytes in size. BigQuery supports"
+                          + " request sizes up to 10MB. You can set the pipeline option"
+                          + " maxStreamingBatchSize to a larger number to unblock this pipeline.",
+                      nextRowSize));
+            }
             errorContainer.add(
```

Comment:
Now about only actively failing for rows that are larger than 10MB (BQ RPC limit) which would surely fail when submitted ? For most users configured "maxRowBatchSize" would be much less than actual BQ limit.